docs/rfcs/xxxx-backwards-compatibility.md

```diff
@@ -0,0 +1,104 @@
+
+  The backwards compatibility rules of these packages follow the same expectations as the [Go 1 compatibility][] expectations.
+
+- **The scope of backwards compatibility**. Backwards compatibility is only defined for major version 1; we reserve the right to change the definition of backwards compatibility between major versions. 
+
+If a breaking change is introduced in a minor change accidentally, and that breaking change is not covered by one of the [exceptions][] defined below, it is a bug. In these cases, a patch release should be introduced to undo the breaking change. 
+
+[exceptions]: #exceptions-to-backwards-compatibility
+[Go 1 compatibility]: https://go.dev/doc/go1compat
+
+### Exceptions to backwards compatibility 
+
+It's impossible to guarantee that full backwards compatibility is achieved. There are some exceptions which may cause a breaking change without a new major version:
+
+- Non-stable functionality: Functionality which is explicitly marked as non-stable are exempt from backwards compatibility between minor releases.
+
+  Non-stable functionality should be backwards compatible between patch releases, unless a breaking change is required for that patch release.
+
+- Security: a breaking change may be made if a security fix requires making a breaking change. 
+
+- Legal requirements: a breaking change may be made if functionality depends on software with a license incompatible with our own.
+
+- Non-versioned network APIs: internal network APIs, such as the internal API used to drive the Flow web UI, are not subject to backwards compatibility guarantees.
+
+- Undocumented behavior: relying on undocumented behavior may break between minor releases. 
+
+- Upstream dependencies: part of the public API of Grafana Agent may directly expose the public API of an upstream dependency. In these cases, if an upstream dependency introduces a breaking change, we may be required to make a breaking change to our public API as well.   
+
+- Other telemetry data: metrics, logs, and traces may change between releases. Only telemetry data which is used in official dashboards is protected under backwards compatibility.
+
+### Avoiding breaking changes 
+
+Backwards compatibility guarantees introduce new challenges while writing code,
+as it means finding creative ways to iterate without making breaking changes
+immediately. Here are some example strategies one could use to avoid making a
+breaking change:
+
+- Keep things internal: the easiest route is to just not expose things publicly. For example, put Go code in `internal/` packages, or use unexported methods or types.
+
+- Rely on deprecations: if a publicly exposed thing must be changed, consider deprecating it.
+
+  - Instead of removing something, consider making it a no-operation and adding a deprecation warning. 
```

Comment:
I'm not sure about this approach, would prefer if we don't include it here and not use it. 

Maybe I'm misunderstanding something, but it seems to me like we could have e.g. `func SendMetrics(...) error` that is being change to `return nil`. In most cases this would break people's setup and hence would violate the:

> can upgrade their version of Grafana Agent without needing to make any changes with the way they interact with Grafana Agent

Reply:
I agree, I already removed some recommendations from the RFC, and I think I should remove the recommendation here too.